docs/adr/0001-the-workflowinstance-change-log.md

```diff
@@ -0,0 +1,44 @@
+# The WorkflowInstance Change Log
+
+## Status
+
+Accepted
+
+## Context
+
+We have recently added a `WorkflowInstance` Change Log to the system.
+
+When the `WorkflowEngine` updates the `WorfklowInstance`, it persists it to storage. In parallel with this it also persists a record to the `IWorkflowInstanceChangeLog`. This consists of the updated state of the `WorkflowInstance` together with the `IWorkflowTrigger` that caused the change (or `null` if this was the original initialization of the instance).
+
+## Decision
+
+The Workflow Instance Change Log should be thought of as a structured log of the changes that have been made to a workflow instance, to be used (with the help of tooling) for manual diagnostic and recovery purposes.
+
+It is not a Transaction Log in the sense of a history of actions executed by a database management system used to guarantee ACID properties over crashes or hardware failures.
+
+## Consequences
+
+### Faulting the `WorkflowEngine`
+
+The `WorkflowEngine` is considered to have faulted if _either_ the workflow instance fails to store, _or_ the change long entry fails to store.
+
+This is _unrelated_ to whether a particular `WorkflowInstance` has entered the `Faulted` state - that is "normal operation" for the Engine, and that class of error is handled gracefully.
```

Comment:
It occurs to me: should we use different words? Workflow engines and workflow instances can both be `Faulted` but these are quite different things. The fact that it was necessary to add an explanation that these are two different things suggests that it might be useful to find a way to distinguish them in our terminology?

As you go on to state, this is somewhat poorly defined right now anyway, so we're probably early enough in our understanding of this that using different terminology would not be disruptive.

It occurs to me that this might not even be a state of the engine. Is it actually just that the workflow engine has failed in its attempt to process this particular operation, but otherwise everything else is probably fine? This is may be a useful distinction because we might want to use this "failed/faulted/whatever" label as a description of what happened with a particular trigger.